src/UI/Components/TreeTable/Helpers/TreeExpansionManager.ts

```diff
@@ -7,23 +7,52 @@ export interface ExpansionState {
 export class TreeExpansionManager {
   constructor(private readonly separator: string) {}
 
-  private createState(keys: string[]): ExpansionState {
-    const pairs = keys.map((id) => [id, false]);
+  private createState(keys: string[], state = false): ExpansionState {
+    const pairs = keys.map((id) => [id, state]);
     return fromEntries(pairs);
   }
 
-  create(keys: string[]): ExpansionState {
-    return this.createState(keys);
+  create(keys: string[], state = false): ExpansionState {
+    return this.createState(keys, state);
   }
 
   toggle(state: ExpansionState, key: string): ExpansionState {
     if (state[key]) return this.close(state, key);
     return this.open(state, key);
   }
 
+  toggleAll(
+    state: ExpansionState,
+    key: string,
+    shouldOpen: boolean
+  ): ExpansionState {
+    if (shouldOpen) return this.openAll(state, key);
+    return this.close(state, key);
+  }
+
   get(state: ExpansionState, key: string): boolean {
     return state[key];
   }
+  checkIfAllChildrenAreOpened(state: ExpansionState, key: string): boolean {
```

Comment:
Are you still using this one somewhere? If not it can be removed I think :D

Reply:
Yeah, you right, I forgot to delete that one and one below right away :D